refactor: share common drt test functions

keivenchang opened the pull request:
#### Overview:

Refactors test helper functions to eliminate code duplication by consolidating shared test utilities into a centralized module and cleaning up unused imports.

#### Details:

- Improved code maintainability by reducing duplication across test modules. I consolidated `create_test_drt_async` and `create_test_drt_with_settings_async` functions from multiple modules into `distributed::test_helpers`
- Removed unused imports from system_status_server.rs (HashMap, body, DefaultMakeSpan, TraceParent)
- Reorganized test modules to clearly separate regular tests from integration tests
- Cleaned up redundant feature gate attributes within integration test modules

#### Where should the reviewer start?

- lib/runtime/src/distributed.rs - Review the new test_helpers module with consolidated test functions
- lib/runtime/src/system_status_server.rs - Check the reorganized test module structure and import cleanup
- lib/runtime/src/metrics.rs - Verify updated imports now use the centralized test helpers


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->

## Summary by CodeRabbit

* **Tests**
  * Centralized test helpers for distributed runtime, reducing duplication.
  * Reorganized integration tests under a single gated module; unit tests remain separate.
  * Simplified configuration gates so some tests run unconditionally.
  * Updated tests to use shared helpers for runtime creation and server info.

* **Chores**
  * Cleaned up imports and minor formatting across test-related code.

No user-facing changes; functionality remains unchanged.

<!-- end of auto-generated comment: release notes by coderabbit.ai -->

keivenchang committed:
refactor: consolidate test helper functions into distributed::test_helpers module

- Move create_test_drt_async and create_test_drt_with_settings_async from metrics and system_status_server modules to centralized location
- Update all test imports to use the consolidated helpers
- Eliminate code duplication across test modules

keivenchang committed:
refactor: clean up imports and reorganize test modules in system_status_server

- Remove unused imports (HashMap, body, DefaultMakeSpan, TraceParent)
- Consolidate std::sync imports
- Separate regular tests from integration tests into distinct modules
- Remove redundant feature gates inside integration_tests module
- Move test command comments to appropriate module locations

rmccorm4 commented:
1. Please address code rabbit feedback
2. Woah! The cargo cache has increased again from ~3.X GB to ~7 GB. I'm guessing this is from recent KVBM merge or something. However, I recently freed up a bunch of disk space in the actions - I don't understand why this still appears to be having disk space issues. CC @nv-anants if you have any ideas here - this is becoming too flaky at this point.
    - referring to this rust check failure: https://github.com/ai-dynamo/dynamo/actions/runs/17112941409/job/48538122781?pr=2583

keivenchang committed:
refactor: simplify and merge duplicate test helper functions

- Simplified create_test_drt_with_settings_async to use Runtime::from_current()
- Merged create_test_drt_with_settings_async into create_test_drt_async
- Both functions were identical after simplification
- Updated all call sites to use the single unified function
- All 188 tests pass with the simplified implementation

keivenchang committed:
chore: remove unused imports from system_status_server integration tests

Removed 9 unused imports:
- crate::logging::tests::load_log
- anyhow::anyhow (kept Result)
- chrono::{DateTime, Utc}
- jsonschema::{Draft, JSONSchema}
- serde_json::Value
- std::fs::File
- std::io::{BufRead, BufReader}
- stdio_override::*
- tokio::time::sleep (kept Duration)

keivenchang commented on lib/runtime/src/system_status_server.rs:
```diff
@@ -381,8 +326,28 @@ mod tests {
+        create_test_drt_async, create_test_drt_with_settings_async,
+    };
+    use crate::logging::tests::load_log;
+    use crate::metrics::MetricsRegistry;
```
Removed imports, thank you.
